### PR TITLE
Raising waiting threads when a thread returns.

### PR DESCRIPTION
**Pull request by Rinnegatamante**

### CODE REVIEW

**Comment by VelocityRa** [comment]
@Rinnegatamante Merge?

**Comment by Rinnegatamante** [comment]
This didn't solve the issue referred in the PR (before these commit, nxengine used to softlock at startup, with this, it softlocks as soon as you try to load Cave Story data).

**Comment by Rinnegatamante** [comment]
Ready to review. It now fully fixes CaveStory regression.

File: src/emulator/kernel/src/thread/thread.cpp [hunk 1]
```diff
@@ -105,6 +105,16 @@ SceUID create_thread(Ptr<const void> entry_point, KernelState &kernel, MemState
     return thid;
 }
 
+void raise_waiting_threads(ThreadState *thread) {
```
Comment (VelocityRa) [inline src/emulator/kernel/src/thread/thread.cpp]: Can you make this function available in `thread_functions.h`?

**Pull request merged**